src/util/triplea/MapXMLCreator/CanalDefinitionsPanel.java

```diff
@@ -0,0 +1,499 @@
+package util.triplea.MapXMLCreator;
+
+import games.strategy.util.Tuple;
+
+import java.awt.Color;
+import java.awt.FontMetrics;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.awt.event.ActionEvent;
+import java.awt.event.MouseEvent;
+import java.awt.geom.Rectangle2D;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import javax.swing.AbstractAction;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+import util.triplea.MapXMLCreator.TerritoryDefinitionDialog.DEFINITION;
+
+/**
+ * 
+ * @author Erik von der Osten
+ * 
+ */
+public class CanalDefinitionsPanel extends ImageScrollPanePanel
+{
+	protected static final String NEW_CANAL_OPTION = "<new Canal>";
+
+	SortedSet<String> m_selectedLandTerritories = new TreeSet<String>();
+	SortedSet<String> m_selectedWaterTerritories = new TreeSet<String>();
+	String m_currentCanalName = null;
+
+	private CanalDefinitionsPanel()
+	{
+		s_instance = this;

```

Comment:
s_instance should be private in the parent class. Accessing variables from of a parent class from a child is a violation of encapsulation. Bottom line, makes it reallly hard to trace code, to know what state variables will be in. Not only do you have to look at and understand the parent class, but you also have to be aware of EVERY child class, and every line of code in each of those child classes. Now to understand one method, you have to look at N classes with M more methods.. This does not make for easy to understand code.